h2o-algos/src/main/java/hex/gam/GAM.java

```diff
@@ -233,6 +233,9 @@ private void validateGamParameters() {
         if (!dataset.vec(cname).isNumeric())
           error("gam_columns", "column " + cname + " is not numerical and cannot be used as a gam" +
                   " column.");
+        if (dataset.vec(cname).isInt() && ((dataset.vec(cname).max() - dataset.vec(cname).min() + 1) < _parms._num_knots[index]))
```

Comment:
@km2018 please move this test to below line 255 where _parms._num_knots is defined if users do not specify it.